### PR TITLE
Add a new field for token inactivity timeout in OAuth TokenConfig

### PR DESCRIPTION
**Pull request by vareti**
This PR adds a new token inactivity timeout field in OAuth TokenConfig. This field is needed for enabling the feature that expires session tokens that are not used within the duration. It accepts string values that can be serialized into `metav1.Duration`.

Needed for https://github.com/openshift/cluster-kube-apiserver-operator/pull/874

### CODE REVIEW

File: config/v1/types_oauth.go [hunk 1]
```diff
@@ -48,7 +48,17 @@ type TokenConfig struct {
...
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// Valid values are integer values:
+	//   x < 0  Tokens time out is enabled but tokens never timeout unless configured per client (e.g. `-1`)
+	//   x = 0  Tokens time out is disabled (default)
+	//   x > 0  Tokens time out if there is no activity for x seconds
+	// The current minimum allowed value for X is 300 (5 minutes)
 	// +optional
 	AccessTokenInactivityTimeoutSeconds int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
```
Comment (deads2k) [inline config/v1/types_oauth.go:63]: Seems like users would prefer to enter a string duration `2d12h5m` as opposed to int seconds.  why not simply retire the field entirely and create that?

**Comment by deads2k** [comment]
/hold

We need an e2e test proving the kube-apiserver actually does the right thing after reacting to this setting.

File: config/v1/types_oauth.go (hunk 1 above)
Comment (vareti) [reply config/v1/types_oauth.go]: @deads2k Added a new field `AccessTokenInactivityTimeout` of type `metav1.Duration`. Please take a look.

**Comment by vareti** [comment]
I will follow-up with an e2e test in operator PR.

File: config/v1/types_oauth.go [hunk 2]
```diff
@@ -51,6 +51,16 @@ type TokenConfig struct {
...
+
+	// accessTokenInactivityTimeout defines the default token
+	// inactivity timeout for tokens granted by any client.
+	// The value represents the maximum amount of time that can occur between
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// Takes valid time duration string such as "5m", "1.5h" or "2h45m".
+	// +optional
```
Comment (deads2k) [inline config/v1/types_oauth.go:62]: tell me what happens if this is not set.

File: config/v1/types_oauth.go [hunk 3]
```diff
@@ -51,6 +51,16 @@ type TokenConfig struct {
 	// accessTokenInactivityTimeoutSeconds - DEPRECATED: setting this field has no effect.
 	// +optional
 	AccessTokenInactivityTimeoutSeconds int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
+
+	// accessTokenInactivityTimeout defines the default token
```
Comment (deads2k) [inline config/v1/types_oauth.go:55]: defines the token inactivity timeout.  If you say default, people will try to find other ways to set it.

File: config/v1/types_oauth.go (hunk 2 above)
Comment (vareti) [reply config/v1/types_oauth.go:63]: added explanation on the behavior.

File: config/v1/types_oauth.go (hunk 3 above)
Comment (vareti) [reply config/v1/types_oauth.go]: Done.

**Comment by deads2k** [comment]
test is here https://github.com/openshift/cluster-kube-apiserver-operator/pull/874/commits/0529b80345fdb02dbbdea45115734a6aed853d4a#diff-655d06d1f4daf5e47c58076174bfd792 and was confirmed manually

/lgtm

**Comment by vareti** [comment]
@deads2k  Thanks for the review!

/hold cancel

File: config/v1/types_oauth.go [hunk 4]
```diff
@@ -51,6 +51,17 @@ type TokenConfig struct {
...
+	// accessTokenInactivityTimeout defines the token inactivity timeout
+	// for tokens granted by any client.
+	// The value represents the maximum amount of time that can occur between
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// If this value is not set, then tokens are valid until their expiry.
+	// Takes valid time duration string such as "5m", "1.5h" or "2h45m".
+	// +optional
+	AccessTokenInactivityTimeout metav1.Duration `json:"accessTokenInactivityTimeout,omitempty"`
```
Comment (sttts) [inline config/v1/types_oauth.go:64]: sorry for the late call: this must be a pointer. If you marshal this struct in the current shape, you get a null in JSON. That's wrong and also not a valid CR object.

**Pull request merged**